src/librustdoc/html/render.rs

```diff
@@ -572,7 +576,16 @@ pub fn run(mut krate: clean::Crate,
         for attr in attrs.lists("doc") {
             match (attr.name_or_empty().get(), attr.value_str()) {
                 ("html_favicon_url", Some(s)) => {
-                    scx.layout.favicon = s.to_string();
+                    let s = s.to_string();
+                    if scx.layout.favicon_path.is_some() {
+                        if !s.is_empty() {
+                            diag.struct_warn("`favicon-path` option has been passed, ignoring \
```

Comment:
Since we're already iterating through libsyntax's attributes here, why not use its span in this warning?